### ISSUE TITLE
Update Firefox extension in the add-on store

### ISSUE THREAD

**Issue by dwhly** [hypothesis/vision#171]

**Comment by tilgovi** [comment]
We have an extension in the add-on repository, but it hasn't yet passed review.

In addition, we know the extension is deficient in other respects.

Double injection issues (hypothesis/h#1688) and embed clobbering (hypothesis/h#1689)
This is a really nasty experience partners, in particular, who might be likely to be testing Hypothesis on their own properties. We don't want to destroy or deploy over a publisher-supplied instance.

Static asset hosting
We can't load bundled resources into iframes in Firefox using resource: URLs. Therefore, we load it from our site. However, that means that we load the newest code there. It is my understanding that our extension has been rejected in the past for being simply an embedding of a app that we host. However, loading bundled code from the extension for the injection into the host page but server-loaded code into the sidebar is a compatibility hazard.

**Comment by BigBlueHat** [comment]
What if we ship a Firefox extension (not the current one) that simply redirects the current page to Via?

The experience is the same. It could even be "shut off" by un-redirecting the current URL.

We'd ship very little code. We could then revisit a *real* Firefox extension in the future that actually did all the annotation stuff itself and simply published the annotations into Hypothes.is...or (which is probably more likely) consider a project that "heavyweight" out of scope for the time being.

Thoughts?

**Comment by csillag** [comment]
> The experience is the same. 

There are some valid use-cases where redirecting via our own proxy is unacceptable, because of confidentiality / data security / copyright reasons.

There is also local data, which can't be siphoned via via.

So this might work in some situation, but doesn't cover all our planned use-cases.

**Comment by BigBlueHat** [comment]
Correct. :smiley: It's not intended to.

**Comment by csillag** [comment]
> > > What if we ship a Firefox extension (not the current one) that simply redirects the current page to Via?

> > So this might work in some situation, but doesn't cover all our planned use-cases.

> Correct. :smiley: It's not intended to.

Well then, we can't ship this as *the* Firefox extension, because it's supposed to cover all our basic use-cases. Are you suggesting to ship this as an alternate / lightweight companion? That might work, but we also must consider the cost of committing ourselves to maintain one more extra deliverable product.

**Comment by dwhly** [comment]
TL;DR:  Why isn't this working in FF now?  https://addons.mozilla.org/en-US/firefox/addon/hypothesis/
Can we update it and get it functional, while we figure out what we need to do to pass review without creating version control issues for ourselves w/ assets that out of sync?

This gets us a fully functioning FF extension with a warning ("hasn't passed review").

The long term solution remains the same, if serving FF users is important, which I think it probably is.  Ship a FF extension that has feature parity, and is discoverable in the addon store.

> Double injection issues (hypothesis/h#1688) and embed clobbering (hypothesis/h#1689)

Lets prioritize and solve these for Chrome (and via?) too.
